File: mobile/src/main/java/br/com/helpdev/velocimetroalerta/ConfigActivityFragment.java
package br.com.helpdev.velocimetroalerta;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.EditTextPreference;
import android.preference.ListPreference;
import android.preference.Preference;
import android.preference.PreferenceFragment;
import android.preference.PreferenceManager;

import br.com.helpdev.velocimetroalerta.dialogs.DialogPrefVelAlertBLModule;

/**
 * Created by Guilherme Biff Zarelli on 05/04/16.
 */
public class ConfigActivityFragment extends PreferenceFragment implements Preference.OnPreferenceChangeListener {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.configs);
        ListPreference listPreference = (ListPreference) findPreference(getString(R.string.pref_intervalo));
        EditTextPreference editTextPreference = (EditTextPreference) findPreference(getString(R.string.pref_intervalo_valor));
        DialogPrefVelAlertBLModule dialogPrefVelAlertModule = (DialogPrefVelAlertBLModule) findPreference(getString(R.string.pref_module_vel_alert));
        editSummary(listPreference);
        editSummary(editTextPreference);
        editSummary(dialogPrefVelAlertModule);
    }


    private void editSummary(Preference preference) {
        preference.setOnPreferenceChangeListener(this);
        SharedPreferences pref = PreferenceManager.getDefaultSharedPreferences(getActivity());
        Object value = pref.getString(preference.getKey(), "");
        onPreferenceChange(preference, value);
    }

    private void editSummary(DialogPrefVelAlertBLModule preference) {
        preference.setOnPreferenceChangeListener(this);
        onPreferenceChange(preference, preference.getSummary());
    }

    @Override
    public boolean onPreferenceChange(Preference preference, Object newValue) {
        preference.setSummary(newValue.toString());
        return true;
    }
}

File: mobile/src/main/java/br/com/helpdev/velocimetroalerta/ConfigActivityFragment.kt
package br.com.helpdev.velocimetroalerta

import android.content.SharedPreferences
import android.os.Bundle
import android.preference.EditTextPreference
import android.preference.ListPreference
import android.preference.Preference
import android.preference.PreferenceFragment
import android.preference.PreferenceManager
import br.com.helpdev.velocimetroalerta.dialogs.DialogHR

import br.com.helpdev.velocimetroalerta.dialogs.DialogPrefVelAlertBLModule

/**
 * Created by Guilherme Biff Zarelli on 05/04/16.
 */
class ConfigActivityFragment : PreferenceFragment(), Preference.OnPreferenceChangeListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        addPreferencesFromResource(R.xml.configs)
        val listPreference = findPreference(getString(R.string.pref_intervalo)) as ListPreference
        val editTextPreference = findPreference(getString(R.string.pref_intervalo_valor)) as EditTextPreference
        val dialogPrefVelAlertModule = findPreference(getString(R.string.pref_module_vel_alert)) as DialogPrefVelAlertBLModule
        val dialogPrefHr = findPreference(getString(R.string.pref_hr_sensor)) as DialogHR
        editSummary(listPreference)
        editSummary(editTextPreference)
        editSummary(dialogPrefVelAlertModule)
        editSummary(dialogPrefHr)
    }


    private fun editSummary(preference: Preference) {
        preference.onPreferenceChangeListener = this
        val pref = PreferenceManager.getDefaultSharedPreferences(activity)
        val value = pref.getString(preference.key, "")
        onPreferenceChange(preference, value)
    }

    private fun editSummary(preference: DialogPrefVelAlertBLModule) {
        preference.onPreferenceChangeListener = this
        onPreferenceChange(preference, preference.summary)
    }

    private fun editSummary(preference: DialogHR) {
        preference.onPreferenceChangeListener = this
        onPreferenceChange(preference, preference.summary)
    }

    override fun onPreferenceChange(preference: Preference, newValue: Any): Boolean {
        preference.summary = newValue.toString()
        return true
    }
}

